README.md

```diff
@@ -210,58 +240,121 @@ Note that this value may be `null` if the request body size is unknown in
 advance because the request message uses chunked transfer encoding.
 
 ```php 
-$http = new Server($socket, function (RequestInterface $request, Response $response) {
+$http = new Server($socket, function (RequestInterface $request) {
     $size = $request->getBody()->getSize();
     if ($size === null) {
-        $response->writeHead(411, array('Content-Type' => 'text/plain'));
-        $response->write('The request does not contain an explicit length.');
-        $response->write('This server does not accept chunked transfer encoding.');
-        $response->end();
-        return;
+        $body = 'The request does not contain an explicit length.';
+        $body .= 'This server does not accept chunked transfer encoding.';
+
+        return new Response(
+            411,
+            array('Content-Type' => 'text/plain'),
+            $body
+        );
     }
-    $response->writeHead(200, array('Content-Type' => 'text/plain'));
-    $response->end("Request body size: " . $size . " bytes\n");
+
+    return new Response(
+        200,
+        array('Content-Type' => 'text/plain'),
+        "Request body size: " . $size . " bytes\n"
+    );
 });
 ```
 
 ### Response
 
-The `Response` class is responsible for streaming the outgoing response body.
+The callback function passed to the constructor of the [Server](#server)
+is responsible to return process a response, which will be delivered to the client.
+This function MUST return either the 
+[PSR-7 ResponseInterface](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-7-http-message.md#33-psrhttpmessageresponseinterface)
+object or a 
+[ReactPHP Promise](https://github.com/reactphp/promise#reactpromise)
+which will resolve a `PSR-7 ResponseInterface` object.
 
-It implements the `WritableStreamInterface`.
+```php 
+$http = new Server($socket, function (RequestInterface $request, Response $response) {
+    $body = "The method of the request is: " . $request->getMethod();
+    $body .= "The requested path is: " . $request->getUri()->getPath();
 
 See also [example #3](examples) for more details.
 
 The constructor is internal, you SHOULD NOT call this yourself.
 The `Server` is responsible for emitting `Request` and `Response` objects.
```

Comment:
This line and the 3 lines above it are no longer relevant and ended up in middle of a code example, see https://github.com/legionth/http/blob/c0aef5da52404f458222acf3554c5f591c63395d/README.md#response